### PR TITLE
Issue 972 csv import - fix2

### PR DESCRIPTION
**Pull request by PawelJastrzebski**
- fix encoding issue https://github.com/quadratichq/quadratic/issues/972
- fix import csv with description before headers https://github.com/quadratichq/quadratic/issues/970

Based on: https://github.com/quadratichq/quadratic/pull/1000 (unable to merge)

### CODE REVIEW

File: quadratic-core/src/controller/operations/import.rs [hunk 1]
```diff
@@ -191,3 +207,12 @@ impl GridController {
         Ok(ops)
     }
 }
+
+fn read_utf16(bytes: &[u8]) -> Option<String> {
```
Comment (ddimaria) [inline quadratic-core/src/controller/operations/import.rs:211]: @PawelJastrzebski can you rewrite this using `safe` code?

File: quadratic-core/tests/csv_import/simple.csv [hunk 2]
```diff
@@ -0,0 +1,11 @@
+city,region,country,population
```
Comment (ddimaria) [inline quadratic-core/tests/csv_import/simple.csv:1]: For sample data used in tests, we're using `/quadratic-rust-shared/data/`

File: quadratic-core/src/controller/operations/import.rs (hunk 1 above)
Comment (PawelJastrzebski) [reply quadratic-core/src/controller/operations/import.rs:211]: I'll look into it

**Comment by PawelJastrzebski** [comment]
@ddimaria 
- update `read_utf16()` to safe rust
- moved test files to `/quadratic-rust-shared/data/`

**Review by ddimaria — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**